extensions/amp-social-share/0.2/social-share.js

```diff
@@ -91,3 +134,64 @@ export function SocialShare(props) {
     </div>
   );
 }
+
+/**
+ * @param {!JsonObject} props
+ * @param {?string} name
+ * @return {!JsonObject}
+ */
+function checkProps(props, name) {
+  const {type, shareEndpoint, params, bindings, width, height} = props;
+
+  // Verify type is provided
+  if (type === undefined) {
+    throw new Error(`The type attribute is required. ${name}`);
+  }
+
+  // bindings and params props must be objects
+  if (params && !isObject(params)) {
+    throw new Error(`The params property should be an object. ${name}`);
+  }
+  if (bindings && !isObject(bindings)) {
+    throw new Error(`The bindings property should be an object. ${name}`);
+  }
+
+  // User must provide shareEndpoint if they choose a type that is not
+  // pre-configured
+  const typeConfig = getSocialConfig(type) || dict();
+  const baseEndpoint = typeConfig['shareEndpoint'] || shareEndpoint;
+  if (baseEndpoint === undefined) {
+    throw new Error(
+      `A shareEndpoint is required if not using a pre-configured type. ${name}`
+    );
+  }
+
+  // Verify width and height are valid integers
+  let checkedWidth = Math.floor(width || DEFAULT_WIDTH);
+  let checkedHeight = Math.floor(height || DEFAULT_HEIGHT);
+  if (!(checkedWidth > 0)) {
```

Comment:
nit: `checkedWidth <= 0`

Reply:
Thanks, I've re-written this seciton, although kept this part with `> 0` because when Math.floor returns NaN, it always return false in a comparison so the recommended update would return a different behavior.